### ISSUE TITLE
Convergence issue across Keras 2.2.4 with TF 1.14.0 and Keras 2.3.1 with TF 2.2.0

### ISSUE THREAD

**Issue by novinsh** [tensorflow/tensorflow#40738]
**System information**
- I am using the Keras functional API directly (not through tf.keras) 
- OS Platform and Distribution: Ubuntu 19.10 with Kernel 5.3.0-59-generic
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on a mobile device: It is on PC (HP laptop)
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): v2.2.0-rc4-8-g2b96f3662b 2.2.0
- Python version: 3.7.7 (default, May  7 2020, 21:25:33)
- Bazel version (if compiling from source): Not compiling from source
- GCC/Compiler version (if compiling from source): Not compiling from source but GCC 7.3.0
- CUDA/cuDNN version: Not using GPU
- GPU model and memory: Not using GPU

**Describe the current behavior**
I have a code that converges on an older version of Keras (2.2.4) while it does not converge on a more recent version (2.3.1).

**Describe the expected behavior**
Convergence is expected to be the same (or at least nearly the same). I expect the validation loss to be approximately the same across different versions of Keras and Tensorflow with the same parameters, model, data, etc.

**Standalone code to reproduce the issue**
https://drive.google.com/file/d/1P6SrCS8HFw2fPYvwrmEuK9tyf6-e5Biv/view?usp=sharing

**PS:**
- I downgraded the Tensorflow on Keras 2.3.1 to TF 1.14.0 so that it would be the same as what it is on my older Keras but did not help with the converges but helped with the speed of training (most probably has to do something with the Eager execution).
- I also tried increasing the epoch, despite the fact that it gets slowly close to the desired loss, still does not get to the same expected loss

In my opinion, there is either a difference in default values across versions. Looking at the API of the methods/functions that I have used from Keras library I do not notice any difference between the default values so it must be something more serious going on.

**Comment by amahendrakar** [comment]
Was able to reproduce the issue, please find the gist of it [here](https://colab.research.google.com/gist/amahendrakar/04e89a2dc4e2d808618d77ca3907f606/40738.ipynb). Thanks!

**Comment by gowthamkpr** [comment]
Please take a look at this [issue1](https://stackoverflow.com/questions/58441514/why-is-tensorflow-2-much-slower-than-tensorflow-1) and also this [issue](https://github.com/tensorflow/tensorflow/issues/33487) let me know if it helps. Thanks!

**Comment by gowthamkpr** [comment]
Closing this issue as it has been inactive for more than 2 weeks. Please add additional comments for us to open this issue again. Thanks!
